packages/create-amplify/src/package-manager-controller/package_manager_controller_factory.test.ts

```diff
@@ -0,0 +1,88 @@
+import { describe, it } from 'node:test';
+import assert from 'assert';
+import { PackageManagerControllerFactory } from './package_manager_controller_factory.js';
+import { type PackageManagerController } from './package_manager_controller.js';
+
+type PackageManagerControllerType = PackageManagerController & {
+  executable: string;
+  binaryRunner: string;
+  addTypescript?: boolean;
+  addLockFile?: boolean;
+  installCommand?: string;
+};
+
+void describe('packageManagerControllerFactory', () => {
+  const packageRoot = '/path/to/project';
+
+  void describe('getPackageManagerController', () => {
+    void it('should return the correct package manager controller for npm', () => {
```

Comment:
Could we turn this into a programmatic test matrix? Something like:
```ts
const testCases = [
  {
    name: 'npm',
    userAgent: 'npm/7.0.0 node/v15.0.0 darwin x64',
    expectedInstanceof: NpmPackageManagerController
  },
  {
    name: 'yarn-modern',
    userAgent: 'yarn/2.0.0 node/v15.0.0 darwin x64',
    expectedInstanceof: YarnModernPackageManagerController
  },
  {
    ... etc
  }
];

for (const testCase of testCases) {
  void it(`returns correct package manager for ${testCase.name}`, () => {
    const packageManagerControllerFactory =
        new PackageManagerControllerFactory(
          packageRoot,
          testCase.userAgent
        );

    const packageManagerController =
        packageManagerControllerFactory.getPackageManagerController();
    
    assert.ok(packageManagerController instanceof testCase.expectedInstanceOf)
  }
}
```

Reply:
Done e98dbb28aa